x-pack/plugins/observability_solution/observability_ai_assistant/server/service/knowledge_base_service/index.ts

```diff
@@ -279,16 +293,23 @@ export class KnowledgeBaseService {
       )
     );
 
-    this.dependencies.logger.info('Processed all queued operations');
+    this.dependencies.logger.info(`Finished processing ${operations.length} queued operations`);
   }
 
   queue(operations: KnowledgeBaseEntryOperation[]): void {
     if (!operations.length) {
       return;
     }
 
-    if (!this.hasSetup) {
-      this._queue.push(...operations);
+    this.dependencies.logger.debug(
+      `Adding ${operations.length} operations to queue. Queue size now: ${this._queue.length})`
+    );
+    this._queue.push(...operations);
```

Comment:
This looks like a bug to me: afaict we were not adding operations to the queue when `isModelReady=true`. This meant that anything imported after the model has been setup won't be stored. 

In general I _really_ hope we can get rid of, or separate the queuing logic **entirely** from the knowledge base. It's deeply embedded making everything around the KB much more complex than it needs to be.